airflow/providers/amazon/aws/operators/sagemaker_base.py

```diff
@@ -41,12 +41,12 @@ class SageMakerBaseOperator(BaseOperator):
     integer_fields = []  # type: Iterable[Iterable[str]]
 
     @apply_defaults
-    def __init__(self, *, config, aws_conn_id='aws_default', **kwargs):
+    def __init__(self, *, config: dict, aws_conn_id: str = 'aws_default', **kwargs):
         super().__init__(**kwargs)
 
         self.aws_conn_id = aws_conn_id
         self.config = config
-        self.hook = None
+        self.hook = SageMakerHook(aws_conn_id=self.aws_conn_id)
```

Comment:
Please don't do this -- hooks can instantiate resources, and so should not be created in operator constructors, but only created when needed.

Either annotate this as a `self.hook: Optional[SageMakerHook] = None`

or

```python


    @cached_property
    def hook(self):
         return SageMakerHook(aws_conn_id=self.aws_conn_id)
```

(you'll need an import for that last one, we already use this elsewhere, aws/base_hook.py for instance.